// vue-onsenui/esm/components/VOnsPopover.js
import 'onsenui/esm/elements/ons-popover.js';
import { hidable, hasOptions, dialog, deriveEvents, deriveDBB, portal } from '../mixins/index.js';
import { openBlock, createElementBlock, normalizeProps, guardReactiveProps, renderSlot } from 'vue';

const name = 'v-ons-popover';

  var script = {
    name,
    mixins: [hidable, hasOptions, dialog, deriveEvents(name), deriveDBB, portal],

    props: {
      target: {
        validator(value) {
          return value.__isVue || typeof value === 'string' || value instanceof Event || value instanceof HTMLElement;
        }
      }
    },

    computed: {
      normalizedTarget() {
        return this.target.$el || this.target;
      },
      normalizedOptions() {
        if (this.target) {
          return {
            target: this.normalizedTarget,
            ...this.options
          };
        }
        return this.options;
      }
    }
  };

function render(_ctx, _cache, $props, $setup, $data, $options) {
  return (openBlock(), createElementBlock("ons-popover", normalizeProps(guardReactiveProps(_ctx.orderedProps)), [
    renderSlot(_ctx.$slots, "default")
  ], 16))
}

script.render = render;

export { script as default };
