// vue-onsenui/esm/components/VOnsSpeedDial.js
import 'onsenui/esm/elements/ons-speed-dial.js';
import { deriveEvents, hidable } from '../mixins/index.js';
import { openBlock, createElementBlock, renderSlot } from 'vue';

const name = 'v-ons-speed-dial';

  var script = {
    name,
    mixins: [deriveEvents(name), hidable],

    emits: ['update:open'],

    props: {
      open: {
        type: Boolean,
        default: undefined
      }
    },

    methods: {
      _shouldUpdate() {
        return this.open !== undefined && this.open !== this.$el.isOpen();
      },
      _updateToggle() {
        this._shouldUpdate() && this.$el[this.open ? 'showItems' : 'hideItems'].call(this.$el);
      }
    },

    watch: {
      open() {
        this._updateToggle();
      }
    },

    mounted() {
      this._updateOpenHandler = () => this._shouldUpdate() && this.$emit('update:open', this.$el.isOpen());

      ['open', 'close'].forEach(event => this.$el.addEventListener(event, this._updateOpenHandler));

      this._updateToggle();
    },

    beforeDestroy() {
      ['open', 'close'].forEach(event => this.$el.removeEventListener(event, this._updateOpenHandler));
    }
  };

function render(_ctx, _cache, $props, $setup, $data, $options) {
  return (openBlock(), createElementBlock("ons-speed-dial", null, [
    renderSlot(_ctx.$slots, "default")
  ]))
}

script.render = render;

export { script as default };
