// vue-onsenui/esm/components/VOnsTab.js
import 'onsenui/esm/elements/ons-tab.js';
import { openBlock, createElementBlock, withModifiers } from 'vue';

var script = {
    name: 'v-ons-tab',
    inject: ['tabbar'],
    emits: ['click'],

    props: {
      page: { },
      props: { },
      active: {
        type: Boolean
      }
    },

    methods: {
      action(event) {
        let runDefault = true;
        this.$emit('click', { preventDefault: () => runDefault = false });

        if (runDefault) {
          this.tabbar.$el.setActiveTab(this.$el.index, { reject: false, ...this.tabbar.options });
        }
      }
    },

    watch: {
      active() {
        this.$el.setActive(this.active);
      }
    }
  };

const _hoisted_1 = ["active"];

function render(_ctx, _cache, $props, $setup, $data, $options) {
  return (openBlock(), createElementBlock("ons-tab", {
    active: $props.active || null,
    onClick: _cache[0] || (_cache[0] = withModifiers((...args) => ($options.action && $options.action(...args)), ["prevent"]))
  }, null, 8, _hoisted_1))
}

script.render = render;

export { script as default };
